aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -382,7 +382,7 @@ def presubmit(self, folder):
         for key, value in job_tmpl.job_resource.items():
             subst_dict[key] = value
         mpi_args = [arg.format(**subst_dict) for arg in computer.get_mpirun_command()]
-        extra_mpirun_params = self.node.get_option('mpirun_extra_params')  # same for all codes in the same calc
+        extra_mpirun_params = self.node.get_option('mpirun_extra_params') or []
```

Comment:
I'm wondering a bit - doesn't it make sense to have the default value specified in the port itself rather than outside?

In the mode above, anyone using the port has to "know" what the default value is.

I.e. I guess in port syntax, `default=lambda: []` is perhaps the way to go...

Reply:
I realized another disadvantage of specifying an empty list or dictionary as default, by doing that, each `CalcJobNode` will always have those attributes, even though they are almost always empty.
```
{
    "environment_variables": {},
    "mpirun_extra_params": [],
}
```
By not specifying "empty" defaults, we prevent these unnecessary attributes. The same actually goes for all other options with "empty" defaults, like `prepend_text` and `append_text`. Thinking about it now, maybe I should not change this behavior in this PR but we should think about this whether this is desirable behavior. I will implement the change you proposed and then open an issue